packages/frontend-2/composables/dates.ts

```diff
@@ -63,15 +68,51 @@ export const formattedFullDate = (date: ConfigType): string =>
  * formattedRelativeDate('2023-12-31') -  returns "1 day ago"
  * formattedRelativeDate('2023-12-31', { prefix: true }) -  returns "1 day ago"
  */
-export const formattedRelativeDate = (
+const formattedRelativeDate = (
   date: ConfigType,
+  now: ConfigType,
   options?: Partial<{ prefix: boolean; capitalize: boolean }>
 ): string => {
   if (options?.prefix) {
-    return isTimeframe(date)
-      ? customRelativeTime(date, options?.capitalize)
-      : `on ${customRelativeTime(date)}`
+    return isTimeframe(date, now)
+      ? customRelativeTime(date, now, options?.capitalize)
+      : `on ${customRelativeTime(date, now)}`
   } else {
-    return customRelativeTime(date, options?.capitalize)
+    return customRelativeTime(date, now, options?.capitalize)
+  }
+}
+
+// Remembering and reusing same now() value in SSR and CSR to avoid hydration mismatches
+const useNowState = () => useState('now', () => new Date())
+
+export const useDateFormatters = () => {
+  const state = useNowState()
+  const { $isAppHydrated } = useNuxtApp()
+
+  return {
+    /**
+     * Formats a given date input into a relative time string with optional prefix
+     * @example
+     * Assuming today is January 1st 2024
+     * formattedRelativeDate('2023-12-01') - returns "Dec 12, 2023"
+     * formattedRelativeDate('2023-12-01', { prefix: true }) - returns "on Dec 12, 2023"
+     * formattedRelativeDate('2023-12-31') -  returns "1 day ago"
+     * formattedRelativeDate('2023-12-31', { prefix: true }) -  returns "1 day ago"
+     */
+    formattedRelativeDate: (
+      date: ConfigType,
+      options?: Partial<{ prefix: boolean; capitalize: boolean; now: ConfigType }>
+    ): string => {
+      // during SSR and hydration use static now, afterwards use a fresh one each time
+      // (unless if specific one fed in)
+      const now = options?.now || ($isAppHydrated.value ? new Date() : state.value)
```

Comment:
@Mikehrn using a static `now` date for SSR and CSR (until hydration is done) fixes relative time hydration mismatches